neuralpredictors/measures/np_functions.py

```diff
@@ -14,15 +21,137 @@ def corr(y1, y2, axis=-1, eps=1e-8, **kwargs):
         y2:      second NumPy array
         axis:    dimension(s) along which the correlation is computed. Any valid NumPy axis spec works here
         eps:     offset to the standard deviation to avoid exploding the correlation due to small division (default 1e-8)
-        **kwargs: passed to final numpy.mean operatoin over standardized y1 * y2
+        **kwargs: passed to final numpy.mean operation over standardized y1 * y2
 
     Returns: correlation array
     """
+
     y1 = (y1 - y1.mean(axis=axis, keepdims=True)) / (y1.std(axis=axis, keepdims=True, ddof=0) + eps)
     y2 = (y2 - y2.mean(axis=axis, keepdims=True)) / (y2.std(axis=axis, keepdims=True, ddof=0) + eps)
     return (y1 * y2).mean(axis=axis, **kwargs)
 
 
+def oracle_corr_corrected(repeated_outputs: ArrayLike) -> np.ndarray:
+    """
+    Compute the corrected oracle correlations per neuron.
+    Note that an unequal number of repeats will introduce bias as it distorts assumptions made about the dataset.
+    Oracle correlation corrected overestimates the true oracle correlation (might need a renaming in the future).
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, neuron responses), or a list containing for each
+            image an array of shape (repeats, neuron responses).
+
+    Returns:
+        array: Corrected oracle correlations per neuron
+    """
+
+    var_noise, var_mean = [], []
+    for output in repeated_outputs:
+        var_noise.append(output.var(axis=0))
+        var_mean.append(output.mean(axis=0))
+    var_noise = np.mean(np.array(var_noise), axis=0)
+    var_mean = np.var(np.array(var_mean), axis=0)
+    return var_mean / np.sqrt(var_mean * (var_mean + var_noise))
+
+
+def oracle_corr(repeated_outputs: ArrayLike) -> np.ndarray:
```

Comment:
Continuing the comment above, this function might best be called `jackknife_oracle`.

Reply:
Done!